### PR TITLE
list of resources and brief description

### PR DESCRIPTION
**Pull request by davidcheung**

### CODE REVIEW

File: docs/resources.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Zero components
+**s3 bucket: Terraform-state:** 
+`<project>-<env>-terraform-state`
+Terraform state s3 bucket is used for storing your terraform state in a terraform [remote backend][tf-remote-backend]
+
+**dynamo db: Terraform statelock**:
+Terraform uses [state lock][tf-remote-state] to ensure actions that could contradict each others do not happen at the same time
```
Comment (bmonkman) [inline docs/resources.md:7]: Maybe just "ensures multiple operators can't attempt to make changes to the infrastructure at the same time" ?

File: docs/resources.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Zero components
+**s3 bucket: Terraform-state:** 
+`<project>-<env>-terraform-state`
+Terraform state s3 bucket is used for storing your terraform state in a terraform [remote backend][tf-remote-backend]
+
+**dynamo db: Terraform statelock**:
+Terraform uses [state lock][tf-remote-state] to ensure actions that could contradict each others do not happen at the same time 
+
+**IAM user: `<project>-ci-user`**:
+This user is given the ECR full-access policy, our intention is you will push your container images to ECR via this IAM
```
Comment (bmonkman) [inline docs/resources.md:10]: This user also has access to assume the kubernetes-admin role, which will let it operate the Kuberenetes cluster.

File: docs/resources.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+**Secret manager: `ci-user-aws-key`**:
+This is a credential key pair generated when creating an IAM user, we then store it in secret manager.
```
Comment (bmonkman) [inline docs/resources.md:13]: These credentials will be put into a CI system then used during the build and deploy process, most likely as environment variables.

File: docs/resources.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+**VPC:**
+Subnet for AWS, all our resources will be locally accessible with each others created in here
+by default everything will be under `10.10.xxx.xxx`, [see details][zero-subnet]
+
+**EKS-cluster**:
+Our main kubernetes stack, core of our infrastructure. Our business logic and application will be deployed here, Kubernetes enables ease of scaling / management of all our microservices. 
+Kubernetes will be high availability according to your [az settings][zero-az]
+
+**EKS: k8s-workers:**
+These are EC2 instances spawned according to your `ami-` image tag, any deployments and pods will be spun up in these EC2 instances. You can scale this up at a later time if your requirement increases
```
Comment (bmonkman) [inline docs/resources.md:24]: Maybe de-emphasize the AMI. The user shouldn't really care about the AMI, as in the vast majority of cases they will just use the one provided by AWS.

File: docs/resources.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+For any ingresses that specify that they need TLS, cert-manager will automatically provision a certificate using Lets Encrypt, and handle renewing it automatically on a regular basis. Alongside external-dns, this allows you to make sure your new domains are always secured using HTTPS. 
+
+To enable cert-manager to create certificate for you using [ClusterIssuer with Zero][zero-cluster-issuer], you would be able to reference it with an annotation in your ingress as follow
+```yml
+## kubernetes ingress annotations
+"cert-manager.io/cluster-issuer": clusterissuer-letsencrypt-production
+## spec > tls > secret_name
+secret_name = "tls-certificate"
+```
+Cert-manager will then create a certificate resource, and use letsencrypt ACME servers to validate and deploy the certificate. Then once all you should should be able to see `kubectl get certificates` as `Ready:true`
```
Comment (bmonkman) [inline docs/resources.md:48]: "Then once all you" -> "Within a minute or so, you" ?

File: docs/resources.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+**EKS: ingress-nginx**:
+Use Nginx as a reverse proxy and load balancer for your cluster. This will create an AWS load balancer (ELB/ALB/NLB) and whenever an ingress is created to route traffic to your application, the controller will make sure the LB is up to date and sending traffic where it needs to go. 
+
+**EKS: External-dns**:
+For any ingresses that are added to route traffic for hosts, external-dns will automatically create DNS records for those hosts and point it to the LB created by the ingress controller. This makes is extremely easy to bring up a new site at a specific domain or subdomain. [How it works][external-dns]
+
+**EKS: Cert-manager**:
+For any ingresses that specify that they need TLS, cert-manager will automatically provision a certificate using Lets Encrypt, and handle renewing it automatically on a regular basis. Alongside external-dns, this allows you to make sure your new domains are always secured using HTTPS. 
+
+To enable cert-manager to create certificate for you using [ClusterIssuer with Zero][zero-cluster-issuer], you would be able to reference it with an annotation in your ingress as follow
```
Comment (bmonkman) [inline docs/resources.md:41]: "To enable cert-manager to create certificates for you using [ClusterIssuer with Zero][zero-cluster-issuer], you would annotate your kubernetes ingress as follows:"

File: docs/resources.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+# you can access the dashboard with the following command
+kubectl get secret -o json -n kubernetes-dashboard $(kubectl get secret -n kubernetes-dashboard | grep dashboard-user-token | awk '{print $1}') | jq -r .data.token | base64 -D | pbcopy && \
+open "http://localhost:8001/api/v1/namespaces/kubernetes-dashboard/services/https:kubernetes-dashboard:/proxy/#/login" && kubectl proxy
+```
+
+**EKS: Metrics-server**
+[A collector][k8s-sigs-metrics] of cluster-wide resource metrics. Used by things like HorizontalPodAutoscaler to determine the current usage of pods. Also allows the `kubectl top` command
+
+**EKS: Cluster-autoscaler**
+Kubernetes deployment monitoring metrics gathered by metrics-server, interacts with AWS autoscaling group to scale the cluster. Scaling the cluster based on resources expenditure and [defined rules][asg-config].
```
Comment (bmonkman) [inline docs/resources.md:65]: "Automatically scales the number of worker nodes in the Kubernetes cluster as utilization changes, interacts with AWS auto scaling groups to scale the cluster. Scales the cluster based on [defined limits][asg-config]."

File: docs/resources.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+**RDS**:
+AWS Relational database services, we provision a postgres DB for your application that is within your VPC within your EKS cluster's worker-security group(accessible by your worker-nodes), the database is also pre-configured to be monitored with cloudwatch.
```
Comment (bmonkman) [inline docs/resources.md:68]: "we provision a postgres DB for your application that is within your VPC and accessible by your EKS worker nodes. The database is also pre-configured to be monitored with cloudwatch."

File: docs/resources.md (hunk 2 above)
Comment (bmonkman) [reply docs/resources.md:10]: And has access to the s3 bucket we create for static assets.

**Comment by bmonkman** [comment]
Thanks a lot for writing all this up, @davidcheung ! This is great!

**Comment by bmonkman** [comment]
A few things related to the diagram:

- iam_account and the little person in the IAM box should be removed, this isn't really useful info (and I'm not sure why it's connected to things like the fluentd role)
- Having "IAM access key" in IAM and "ci-user-aws-key" in secret manager is probably redundant, maybe remove the one in IAM?
- The ci user should be connected to the kubernetes-admin role
- There shouldn't be an IAM role for `alb-ingress-controller`
- There are some deployments missing for new things like cert-manager
- I'm not sure showing configmaps provides a lot of value, it's just application-specific configuration related to components that are already being shown on the diagram..

**Comment by davidcheung** [comment]
the helm stuff [cert-manager, clusterIssuer, autoscaler]  i actually categorized them as custom resources(at the bottom-right), I guess they're really just deployments

**Comment by bmonkman** [comment]
There are both custom resources and deployments

**Comment by bmonkman** [comment]
What do you think about another box for the auto scaling group, to show how the cluster-autoscaler manages the ASG?

**Comment by davidcheung** [comment]
AWS autoscaling group itself is not in the VPC right? 
realized i didnt have the metrics server either, added that

**Comment by bmonkman** [comment]
Yes, the ASG can be shown as in the VPC. Technically it's not related to a VPC directly itself, but it specifies the subnets to bring up instances in, and the subnets/instances are within the VPC.

**Comment by bmonkman** [comment]
Fine to leave it as-is though, doesn't really make a difference.

**Pull request merged**